### PR TITLE
feat: adding e2e test for article commenting

### PR DESCRIPTION
**Pull request by JohnAllenTech**
# ✨ Codu Pull Request 💻

Improves E2E testing for the project by adding tests around a user commenting on an article

## Pull Request details

- Checks authenticated user can post a comment
- Checks that an unauthenticated user cannot post a comment

## Any Breaking changes

- None

## Associated Screenshots

image
Tests passing

image
Example comment generated by E2E test. The uuid needs to be added as if multiple a user runs the test suite multiple times we want to always ensure we are finding the comment created for that specific test run

### CODE REVIEW

File: e2e/articles.spec.ts [hunk 1]
```diff
@@ -185,4 +204,22 @@ test.describe("Authenticated Articles Page", () => {
...
+    await expect(page.getByText("Read Full Article").first()).toBeVisible();
+    page.getByText("Read Full Article").first().click();
+    await page.waitForURL(/^http:\/\/localhost:3000\/articles\/.*$/);
+
+    await expect(page.getByPlaceholder("What do you think?")).toBeVisible();
+    await page.getByPlaceholder("What do you think?").fill(commentContent);
+    await page.getByRole("button", { name: "Submit" }).click();
+
+    await expect(page.getByText(commentContent)).toBeVisible();
+  });
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts]: Your wrong

File: e2e/articles.spec.ts [hunk 2]
```diff
@@ -57,6 +58,24 @@
...
     await expect(page.getByText("Code Of Conduct")).toBeVisible();
   });
+
+  test("Should not be able to post a comment on an article", async ({
+    page,
+  }) => {
+    await page.goto("http://localhost:3000");
+    // Waits for articles to be loaded
+    await expect(page.getByText("Read Full Article").first()).toBeVisible();
+    page.getByText("Read Full Article").first().click();
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts]: Will resolve this in next commit

File: e2e/articles.spec.ts [hunk 3]
```diff
@@ -185,4 +204,22 @@
...
+
+  test("Should post a comment on an article", async ({
+    page,
+    browserName,
+  }, workerInfo) => {
+    const commentContent = `This is a great read. Thanks for posting! Sent from ${workerInfo.project.name} + ${randomUUID()}`;
+    await page.goto("http://localhost:3000");
+    // Waits for articles to be loaded
+    await expect(page.getByText("Read Full Article").first()).toBeVisible();
+    page.getByText("Read Full Article").first().click();
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply e2e/articles.spec.ts]: Will resolve this in next commit

**Review by NiallJoeMaher — APPROVED** [review]
🌮🌮🌮🌮

**Pull request merged**